components/net/local_directory_listing.rs

```diff
@@ -0,0 +1,405 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use std::collections::BTreeMap;
+use std::ffi::OsString;
+use std::fs::{DirEntry, Metadata};
+use std::io;
+use std::path::PathBuf;
+
+use embedder_traits::resources::{read_string, Resource};
+use headers::{ContentType, HeaderMapExt};
+use net_traits::request::Request;
+use net_traits::response::{Response, ResponseBody};
+use net_traits::{NetworkError, ResourceFetchTiming};
+use servo_config::pref;
+use servo_url::ServoUrl;
+use time_03::OffsetDateTime;
+use url::Url;
+
+pub fn is_request_allowed(request: &Request) -> Result<(), Response> {
+    if !pref!(network.local_directory_listing.enabled) {
+        // If you want to be able to browse local directories, configure Servo prefs so that
+        // "network.local_directory_listing.enabled" is set to true.
+        Err(Response::network_error(NetworkError::Internal(
+            "Local directory listing feature has not been enabled in preferences".into(),
+        )))
+    } else if !request.origin.is_opaque() {
+        // Checking for an opaque origin as a shorthand for user activation
+        // as opposed to a request originating from a script.
+        // TODO: Raise and link to GitHub issue to request consideration of this concern.
```

Comment:
You could open the issue yourself and link it here.

Reply:
Opened [new issue](https://github.com/servo/servo/issues/32534) and added the number into the TODO comment within the code.